org.adempiere.base/src/org/adempiere/util/ProcessUtil.java

```diff
@@ -150,6 +150,10 @@ public static boolean startJavaProcess(Properties ctx, ProcessInfo pi, Trx trx,
 	 * @return true if process completed successfully
 	 */
 	public static boolean startJavaProcess(Properties ctx, ProcessInfo pi, Trx trx, boolean managedTrx, IProcessUI processMonitor) {
+		if (trx == null) {
+			throw new IllegalArgumentException("Transaction (trx) must not be null");
+		}
+

```

Comment:
I goes through the original code again and the only circumstances a null trx will cause error is for jasper report. We should just fix that and maintain the existing contract that trx can be null (i.e change the line `MProcess proc = new MProcess(ctx, pi.getAD_Process_ID(), trx.getTrxName());` to 'MProcess proc = MProcess.get(pi.getAD_Process_ID());`)

Reply:
done